### ISSUE TITLE
Non-reproducible results with ne120 on cori-knl

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/scream#1608]
After the changes to the "checks" in recent PR's that allow ne120 to run further (still using repo from April27), the first attempt ran 12 steps and failed while trying to write snapshot (different issue). Requesting no snapshots and just trying to see how far I can run, I'm seeing some of the attempts fail (with no good error message) at different points.  Some fail at step 2, others after step 14.

I'm seeing this with both intel and gnu compilers. 
I just tried again with most recent Intel compiler (due to another issue) and verified same sort of behavior.

I'm trying to sort out what the error message is -- in some cases I only see srun error/terminate lines.

But in one gnu case and one intel case, I see:

```
gnu OPT, first case runs 2 steps and fails without useful error, 2nd case runs 14 steps and stops with:

 840: terminate called after throwing an instance of 'std::logic_error'
 840:   what():  /global/cfs/cdirs/e3sm/ndk/se02-apr27/components/scream/src/share/atm_process/atmosphere_process.cpp:282: FAIL:
 840: false
 840: Error! Failed post-condition check (cannot be repaired).
 840:   - Atm process name: Dynamics
 840:   - Property check name: T_mid within interval [140, 500]
 840:   - Atmosphere process MPI Rank: 840
 840:   - Error message: FieldWithinIntervalCheck failed; min = 121.193017; max = 306.649297

Intel OPT, also the first case runs 2 steps without useful error, 2nd attempt runs 14 steps and stops:

 840: terminate called after throwing an instance of 'std::logic_error'
 840:   what():  /global/cfs/cdirs/e3sm/ndk/se02-apr27/components/scream/src/share/atm_process/atmosphere_process.cpp:287: FAIL:
 840: false
 840: Error! Failed post-condition check (cannot be repaired).
 840:   - Atm process name: Dynamics
 840:   - Property check name: T_mid within interval [140, 500]
 840:   - Atmosphere process MPI Rank: 840
 840:   - Error message: FieldWithinIntervalCheck failed; min = 121.191448; max = 306.648456

gnu DEBUG ran 11 steps and fails with:

 840: terminate called after throwing an instance of 'std::logic_error'
 840:   what():  /global/cfs/cdirs/e3sm/ndk/se02-apr27/components/scream/src/share/atm_process/atmosphere_process.cpp:282: FAIL:
 840: false
 840: Error! Failed post-condition check (cannot be repaired).
 840:   - Atm process name: Dynamics
 840:   - Property check name: T_mid within interval [140, 500]
 840:   - Atmosphere process MPI Rank: 840
 840:   - Error message: FieldWithinIntervalCheck failed; min = 121.193006; max = 306.649295
 840:

intel DEBUG also runs 11 steps and fails with:

 840: terminate called after throwing an instance of 'std::logic_error'
 840:   what():  /global/cfs/cdirs/e3sm/ndk/se02-apr27/components/scream/src/share/atm_process/atmosphere_process.cpp:287: FAIL:
 840: false
 840: Error! Failed post-condition check (cannot be repaired).
 840:   - Atm process name: Dynamics
 840:   - Property check name: T_mid within interval [140, 500]
 840:   - Atmosphere process MPI Rank: 840
 840:   - Error message: FieldWithinIntervalCheck failed; min = 121.193022; max = 306.649298

```

**Comment by AaronDonahue** [comment]
We see this temperature error eventually pop up in our ne30 runs as well.  So the cause may be related.

**Comment by ndkeen** [comment]
Just noting that we kinda have 2 issues here -- one is the error itself that apparently we also see with ne30, but also as the title indicates, I don't like it when identical cases do something different. WIll try again with updated repo soon.

**Comment by ndkeen** [comment]
I'm now using the new CPU-only nodes of PM and a master of May 6th. When I run ne120, it fails after 12th step (in a similar way I think) and has been reproducible so far. In fact two runs have identical homme_atm.log files. I could close this issue since we will probably move away from Cori

**Comment by ndkeen** [comment]
With May20th master, I ran 2 cases at ne120 (with SPA) on cori-knl. One with 128 nodes and the other with 256 nodes. Both using 16x4 and both compiled optimized with GNU. After 24 steps, both have the same values in homme_atm.log, which I think means BFB.  I'm trying to resubmit in both cases to be sure, but then I think we can close this issue.

**Comment by ndkeen** [comment]
Unfortunately we don't know what changed to resolve this, but I can't reproduce.
